xbmc/Application.cpp

```diff
@@ -2846,6 +2839,12 @@ bool CApplication::Cleanup()
     delete m_network;
     m_network = NULL;
 
+    if (m_ServiceManager)
```

Comment:
This should not be needed, deinit should be the destructor unless we specifically need to deinit and reinit during execution